.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,31 @@
+name: shellcheck
+on:
+  pull_request:  
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'     
+        required: true
+        default: 'warning'
+
+jobs:
+  shellcheck:
+    runs-on: gardenlinux-amd64
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        echo "Install shellcheck"
+        echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list \
+            &&  sudo apt update
+        sudo apt install shellcheck
+    - run: shellcheck --version
+    - run: |
+        echo "Do shellcheck on all *.sh files"
+        status=0
+        set +e 
+        while IFS= read -r -d '' file
+        do
+            shellcheck --severity=warning "$file"  
+            status=$((status +$?))
+        done < <(find . -name '*.sh' -print0)
```

Comment:
won't this miss a lot of files? most of our scripts don't end in `.sh` so we'll probably need a more clever way to find all.... maybe find all files with executable bit set and then check the `#!` line 🤔

Reply:
good point